pkg/bootstrapper/bootstrapper.go

```diff
@@ -641,9 +679,12 @@ func (wmcb *winNodeBootstrapper) refreshKubeletService(config mgr.Config) error
 	if err := wmcb.service.svcObj.UpdateConfig(config); err != nil {
 		return fmt.Errorf("error updating kubelet service: %v", err)
 	}
-
-	if err := wmcb.startService(); err != nil {
-		return fmt.Errorf("error starting kubelet service: %v", err)
+	// starting the services in dependency list will also start the services they are dependent on
+	for _, dependentSvc := range wmcb.service.svcDependents {
+		wmcb.service.setService(dependentSvc)
+		if err := wmcb.startService(); err != nil {
+			return fmt.Errorf(" error starting dependent service %s for kubelet service", dependentSvc)
+		}
```

Comment:
For each service you find, you should be instantiating a new service object and calling `start` on it. This again looks a bit odd to me otherwise.